core/src/stages/compute_ik.cpp

```diff
@@ -352,17 +353,39 @@ void ComputeIK::compute() {
 		sandbox_scene->getCurrentState().copyJointGroupPositions(jmg, compare_pose);
 
 	double min_solution_distance = props.get<double>("min_solution_distance");
+	bool maximize_clearance = props.get<bool>("maximize_clearance");
+	// disable collision checking between links not in current group in order to compute
+	// clearance between links in current group and rest of the links
+	auto acm = sandbox_scene->getAllowedCollisionMatrix();
+	if (maximize_clearance) {
+		auto& all_link_names = sandbox_state.getRobotModel()->getLinkModelNames();
+		auto& group_link_names = jmg->getLinkModelNames();
+		std::vector<std::string> non_group_links;
+
+		for (const auto& link : all_link_names)
+			if (std::find(group_link_names.begin(), group_link_names.end(), link) == group_link_names.end())
+				non_group_links.push_back(link);
+		acm.setEntry(non_group_links, non_group_links, true);
+	}
 
 	IKSolutions ik_solutions;
-	auto isValid = [sandbox_scene, ignore_collisions, min_solution_distance, &ik_solutions](
+	auto isValid = [sandbox_scene, ignore_collisions, min_solution_distance, maximize_clearance, acm, &ik_solutions](
 	    robot_state::RobotState* state, const robot_model::JointModelGroup* jmg, const double* joint_positions) {
 		for (const auto& sol : ik_solutions) {
-			if (jmg->distance(joint_positions, sol.data()) < min_solution_distance)
+			if (jmg->distance(joint_positions, sol.first.data()) < min_solution_distance)
 				return false;  // too close to already found solution
 		}
 		state->setJointGroupPositions(jmg, joint_positions);
 		ik_solutions.emplace_back();
-		state->copyJointGroupPositions(jmg, ik_solutions.back());
+		state->copyJointGroupPositions(jmg, ik_solutions.back().first);
+		if (maximize_clearance) {
```

Comment:
Do you mind moving the computation of clearance down to cost calculation?
This will simplify the code (e.g. no need to store the clearance in the solutions vector)
and facilitate later migration into a separate cost function.